### PR TITLE
Update code example to match new sdl2.nim syntax

### PR DESCRIPTION
**Pull request by ldhieu**
I've updated the code example for hot-code reloading to match the new sdl2 syntax at [nim-lang/sdl2](https://github.com/nim-lang/sdl2/blob/master/src/sdl2.nim)

I've also added a reminder for people who encounter the error `dlopen(libnimhcr.dylib, 2): image not found` as I did earlier today.

Please review and tell me if anything is amiss. This is my first time doing this.

Cheers,
Hieu

### CODE REVIEW

File: doc/hcr.rst [hunk 1]
```diff
@@ -26,59 +26,61 @@ To install SDL2 you can use ``nimble install sdl2``.
...
-  var posX = 1
-  var posY = 0
-  var dX = 1
-  var dY = 1
-
+  
+  var posX : cint = 1
+  var posY : cint = 0
+  var dX : cint = 1
+  var dY : cint = 1
```
Comment (alaviss) [inline doc/hcr.rst]:
```suggestion
  var posX: cint = 1
  var posY: cint = 0
  var dX: cint = 1
  var dY: cint = 1
```

Some minor cosmetic changes.

File: doc/hcr.rst [hunk 2]
```diff
@@ -26,59 +26,61 @@ To install SDL2 you can use ``nimble install sdl2``.
...
+    while pollEvent(evt):
+      if evt.kind == QuitEvent:
         runGame = false
         break
-      if evt.kind == KEY_DOWN:
-        if evt.key.keysym.scancode == SCANCODE_ESCAPE: runGame = false
-        elif evt.key.keysym.scancode == SCANCODE_F9:
+      if evt.kind == KeyDown:
+        if evt.key.keysym.scancode == getScancodeFromName("Escape"): runGame = false
+        elif evt.key.keysym.scancode == getScancodeFromName("F9"):
```
Comment (alaviss) [inline doc/hcr.rst]: Looks like you can still use the constants, they have `SDL_` prefixes: https://github.com/nim-lang/sdl2/blob/46aa2c0e30e9edf6028bc432a3ed70e51a0fe9f9/src/sdl2/private/keycodes.nim#L4

File: doc/hcr.rst [hunk 3]
```diff
@@ -26,59 +26,61 @@ To install SDL2 you can use ``nimble install sdl2``.
 .. code-block:: nim
 
   # logic.nim
-  import sdl2/sdl
-
+  import sdl2
+
```
Comment (alaviss) [inline doc/hcr.rst]: Can you cleanup the trailing whitespaces?

File: doc/hcr.rst [hunk 4]
```diff
@@ -26,59 +26,61 @@ To install SDL2 you can use ``nimble install sdl2``.
...
-
-    var rect: Rect(x: posX - 25, y: posY - 25, w: 50, h: 50)
-    discard renderer.renderFillRect(rect.addr)
+  
+    discard renderer.setDrawColor(0, 0, 255, 255)
+    discard renderer.clear()
+    discard renderer.setDrawColor(255, 128, 128, 0)
+  
+    var rect: Rect
+    rect = (x: posX - 25, y: posY - 25, w: 50.cint, h: 50.cint)
```
Comment (alaviss) [inline doc/hcr.rst]:
```suggestion
    var rect: Rect = (x: posX - 25, y: posY - 25, w: 50.cint, h: 50.cint)
```

File: doc/hcr.rst [hunk 5]
```diff
@@ -26,59 +26,61 @@ To install SDL2 you can use ``nimble install sdl2``.
...
+    discard renderer.fillRect(rect.addr)
```
Comment (alaviss) [inline doc/hcr.rst]:
```suggestion
    discard renderer.fillRect(rect)
```

There's a variant that takes a `var Rect`.

File: doc/hcr.rst [hunk 6]
```diff
@@ -187,7 +189,9 @@ Native code targets
 Native projects using the hot code reloading option will be implicitly
 compiled with the `-d:useNimRtl` option and they will depend on both
 the ``nimrtl`` library and the ``nimhcr`` library which implements the
-hot code reloading run-time.
+hot code reloading run-time. Both libraries can be found in the ``lib``
+folder of Nim and can be compiled to a ``.dylib`` file to satisfy compiling
+of the example code above.
```
Comment (alaviss) [inline doc/hcr.rst]: I think this should describe how to compile them instead.

**Comment by ldhieu** [comment]
Hello, thank you for your comments @alaviss ! I've added the changes and noted you in the commit message.

Let me know if something else is needed !

Hieu

File: doc/hcr.rst (hunk 1 above)
Comment (ldhieu) [reply doc/hcr.rst]: commit `daccd8464b4186013ef5b59d13b3939512e2fdc1` resolved this

File: doc/hcr.rst (hunk 2 above)
Comment (ldhieu) [reply doc/hcr.rst]: Thanks! I see them now. Commit `daccd8464b4186013ef5b59d13b3939512e2fdc1` resolved this.

File: doc/hcr.rst (hunk 3 above)
Comment (ldhieu) [reply doc/hcr.rst]: I've cleaned up the whitespaces in commit `daccd8464b4186013ef5b59d13b3939512e2fdc1` resolved this

File: doc/hcr.rst (hunk 5 above)
Comment (ldhieu) [reply doc/hcr.rst]: Cheers alaviss
commit `daccd8464b4186013ef5b59d13b3939512e2fdc1` resolved this

File: doc/hcr.rst (hunk 6 above)
Comment (ldhieu) [reply doc/hcr.rst]: I've added some example console commands on UNIX/MacOS

I was not sure whether to use RST `::` or use `.. code-block:: console` (used the former as it's how it's written in this document)

commit `daccd8464b4186013ef5b59d13b3939512e2fdc1` is a suggestion on this.

File: doc/hcr.rst [hunk 7]
```diff
@@ -187,7 +188,25 @@ Native code targets
 Native projects using the hot code reloading option will be implicitly
 compiled with the `-d:useNimRtl` option and they will depend on both
 the ``nimrtl`` library and the ``nimhcr`` library which implements the
-hot code reloading run-time.
+hot code reloading run-time. Both libraries can be found in the ``lib``
+folder of Nim and can be compiled to a ``.dylib`` file to satisfy compiling
```
Comment (alaviss) [inline doc/hcr.rst]: Please note that `lib<name>.dylib` is the format for macOS. This differs between operating systems (`<name>.dll` on Windows and `lib<name>.so` on *nix).

I think we can just get away with "compiled into dynamic libraries".

> satisfy compiling of the example code above.

It's actually only needed at runtime.

Comment (ldhieu) [reply doc/hcr.rst]: Thank you ! Absolutely right.

I've committed `6b31eb153e29ed670141689f71e70d8ef5aeba79` to resolve this.

**Comment by Araq** [comment]
Note: I need to test this manually before merging.

**Pull request merged**